src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindVectorProvidersOperation.java

```diff
@@ -0,0 +1,74 @@
+import io.stargate.sgv2.jsonapi.api.model.command.CommandStatus;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.QueryExecutor;
+import io.stargate.sgv2.jsonapi.service.embedding.configuration.PropertyBasedEmbeddingServiceConfig;
+import io.stargate.sgv2.jsonapi.service.operation.model.Operation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Operation that list all available vector providers into the {@link
+ * CommandStatus#EXISTING_VECTOR_PROVIDERS} command status.
+ */
+public record FindVectorProvidersOperation(
+    PropertyBasedEmbeddingServiceConfig propertyBasedEmbeddingServiceConfig) implements Operation {
+  @Override
+  public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
+    return Uni.createFrom()
+        .item(
+            () -> {
+              // TODO: get all available vector providers
+              Map<String, ReturnVectorProviderConfig> vectorProviders = new HashMap<>();
+              if (propertyBasedEmbeddingServiceConfig.google().enabled()) {
+                vectorProviders.put(
+                    "vertexai",
+                    ReturnVectorProviderConfig.provider(
+                        (propertyBasedEmbeddingServiceConfig.google())));
+              }
+              return new Result(vectorProviders);
+            });
+  }
+
+  // simple result wrapper
+  private record Result(Map<String, ReturnVectorProviderConfig> vectorProviders)
+      implements Supplier<CommandResult> {
+
+    @Override
+    public CommandResult get() {
+      Map<CommandStatus, Object> statuses =
+          Map.of(CommandStatus.EXISTING_VECTOR_PROVIDERS, vectorProviders);
+      return new CommandResult(statuses);
+    }
+  }
+
+  private record ReturnVectorProviderConfig(
+      String url,
+      List<String> supportedAuthentication,
+      List<PropertyBasedEmbeddingServiceConfig.VectorProviderConfig.ParameterConfig> parameters,
+      List<ReturnModelConfig> models) {
+    private static ReturnVectorProviderConfig provider(
+        PropertyBasedEmbeddingServiceConfig.VectorProviderConfig vectorProviderConfig) {
+      ArrayList<ReturnModelConfig> modelsRemoveProperties = new ArrayList<>();
+      for (PropertyBasedEmbeddingServiceConfig.VectorProviderConfig.ModelConfig model :
+          vectorProviderConfig.models()) {
+        ReturnModelConfig returnModel = new ReturnModelConfig(model.name(), model.parameters());
+        modelsRemoveProperties.add(returnModel);
+      }
+      return new ReturnVectorProviderConfig(
+          vectorProviderConfig.url(),
+          vectorProviderConfig.supportedAuthentication(),
+          vectorProviderConfig.parameters(),
+          modelsRemoveProperties);
+    }
+  }
+
+  private record ReturnModelConfig(
+      String name,
+      List<PropertyBasedEmbeddingServiceConfig.VectorProviderConfig.ParameterConfig> parameters) {}
+}
```

Comment:
As discussed in the meeting, we want to remove the properties config. So I didn't directly use the `VectorProviderConfig` to construct the return body. 
 The current return body looks like:
 ```
{
    "status": {
        "vectorProviders": {
            "vertexai": {
                "url": "https://us-central1-aiplatform.googleapis.com/v1/projects/{PROJECT_ID}/locations/us-central1/publishers/google/models/{MODEL}:predict",
                "supportedAuthentication": [
                    "HEADER",
                    "SHARED_SECRETS"
                ],
                "parameters": [
                    {
                        "name": "PROJECT_ID",
                        "type": "STRING",
                        "defaultValue": null,
                        "help": null,
                        "required": true
                    }
                ],
                "models": [
                    {
                        "name": "textembedding-gecko@003",
                        "parameters": [
                            {
                                "name": "autoTruncate",
                                "type": "BOOLEAN",
                                "defaultValue": "true",
                                "help": null,
                                "required": false
                            }
                        ]
                    }
                ]
            }
        }
    }
}
```
If we want to include all the configs, the return body will look like:
```
{
    "status": {
        "vectorProviders": {
            "vertexai": {
                "url": "https://us-central1-aiplatform.googleapis.com/v1/projects/{PROJECT_ID}/locations/us-central1/publishers/google/models/{MODEL}:predict",
                "enabled": true,
                "properties": {
                    "max-retries": "3",
                    "retry-delay-ms": "100",
                    "timeout-ms": "1000"
                },
                "parameters": [
                    {
                        "name": "PROJECT_ID",
                        "type": "STRING",
                        "defaultValue": null,
                        "help": null,
                        "required": true
                    }
                ],
                "supportedAuthentication": [
                    "HEADER",
                    "SHARED_SECRETS"
                ],
                "models": [
                    {
                        "name": "textembedding-gecko@003",
                        "properties": {
                            "max-tokens": "3072"
                        },
                        "parameters": [
                            {
                                "name": "autoTruncate",
                                "type": "BOOLEAN",
                                "defaultValue": "true",
                                "help": null,
                                "required": false
                            }
                        ]
                    }
                ]
            }
        }
    }
}

```

Reply:
I think that `"status"` is just wrong: and I am not sure we even need `"vectorProviders"` level there. So why not return a JSON Object with name as key (so remove 2 layers)?

Or, if we have to conform to shared structure, maybe main level key should be `"data"`?

Reply:
Solved, `status` is from the convention of `findCollections` and `findNamespaces`, we will follow this